doc/sphinx/Pacemaker_Development/c.rst

```diff
@@ -341,6 +341,83 @@ the header code. For example:
     */
 
 
+.. index::
+   pair: C; includes
+
+Includes
+########
+
+Includes should be listed in the following order:
+
+* ``<crm_internal.h>`` must always come first.
+
+* System includes, such as ``<stdbool.h>``, ``<sys/types.h>``, etc.
+
+* Other libraries, such as ``<glib.h>``, ``<libxml/tree.h>``, etc.
+
+* Pacemaker libraries, such as ``<crm/common/results.h>``, ``<pacemaker-internal.h>``,
+  etc.
+
+* Local includes, which are basically anything that would use double quotes
+  instead of brackets as part of the ``#include`` line.
+
+Within each block, list headers alphabetically unless there is a reason not to
+do so.  Put a blank line between each block of includes.
+
+Each include should have a comment listing what symbols from that header are
+being used.  Exceptions to this are:
+
+* Includes that are completely obvious (for example, ``<stdbool.h>`` obviously
+  exports the ``true``, ``false``, and ``bool`` symbols).
+
+* Includes that we use a large number of symbols from (for example, a source
+  file might use a lot of symbols from GLib, all of which are namespaced to
+  start with ``g_`` and are therefore obvious).
+
+* Anything that comes from an internal Pacemaker header since most source
+  files will use a large number of internal symbols.
+
+Symbols can sometimes be included from multiple headers.  This is especially
+the case with fundamental C symbols like ``NULL`` or ``size_t`` which the
+standard defines in several headers.  In this case, use your best judgement
+and try to follow what other files in the same library or tool are doing.
+
+In general, a source file should have a complete list of includes that cover
```

Comment:
I think we should add a paragraph or so about header files as well. We don't include `crm_internal.h` in other header files, for example, but we may include other internal headers. And obviously, "top-level" headers do include subheaders (like `xml.h` including `xml_io.h` and `xml_internal.h` including `xml_io_internal.h`).

Can be a later PR.